examples/component-webhooks/capabilities/hook.ts

```diff
@@ -0,0 +1,134 @@
+  name: "example-webhook",
+  description:
+    "A simple example capability to show how webhooks work with Zarf package deployments.",
+  namespaces: ["zarf"],
+});
+
+const { When } = Webhook;
+
+When(a.Secret)
+  .IsCreatedOrUpdated()
+  .InNamespace("zarf")
+  .WithLabel("package-deploy-info")
+  .Mutate(request => {
+    const secret = request.Raw;
+    let secretData;
+    let secretString: string;
+    let manuallyDecoded = false;
+
+    // Pepr does not decode/encode non-ASCII characters in secret data: https://github.com/defenseunicorns/pepr/issues/219
+    try {
+      secretString = atob(secret.data.data);
+      manuallyDecoded = true;
+    } catch (err) {
+      secretString = secret.data.data;
+    }
+
+    // Parse the secret object
+    try {
+      secretData = JSON.parse(secretString);
+    } catch (err) {
+      throw new Error(`Failed to parse the secret.data.data: ${err}`);
+    }
+
+    for (const deployedComponent of secretData?.deployedComponents ?? []) {
+      if (deployedComponent.status === "Deploying") {
+        Log.info(
+          `The component ${deployedComponent.name} is currently deploying`,
+        );
+
+        const componentWebhook =
+          secretData.componentWebhooks?.[deployedComponent?.name]?.[
+            "test-webhook"
+          ];
+
+        // Check if the component has a webhook running for the current package generation
+        if (componentWebhook?.observedGeneration === secretData.generation) {
+          Log.debug(
+            `The component ${deployedComponent.name} has already had a webhook executed for it. Not executing another.`,
+          );
+        } else {
+          // Seed the componentWebhooks map/object
+          if (!secretData.componentWebhooks) {
+            secretData.componentWebhooks = {};
+          }
+
+          // Update the secret noting that the webhook is running for this component
+          secretData.componentWebhooks[deployedComponent.name] = {
+            "test-webhook": {
+              "name": "test-webhook",
+              "status": "Running",
+              "observedGeneration": secretData.generation,
+            },
+          };
+
+          // Call an async function that simulates background processing and then updates the secret with the new status when it's complete
+          sleepAndChangeStatus(secret.metadata.name, deployedComponent.name);
+        }
+      }
+    }
+
+    if (manuallyDecoded === true) {
+      secret.data.data = btoa(JSON.stringify(secretData));
+    } else {
+      secret.data.data = JSON.stringify(secretData);
+    }
+  });
+
+// sleepAndChangeStatus sleeps for the specified duration and changes the status of the 'test-webhook' to 'Succeeded'.
+async function sleepAndChangeStatus(secretName: string, componentName: string) {
+  await sleep(10);
+
+  // Configure the k8s api client
+  const kc = new k8s.KubeConfig();
+  kc.loadFromDefault();
+  const k8sCoreApi = kc.makeApiClient(k8s.CoreV1Api);
+
```

Comment:
prob should make a tech debt issue to swap to https://github.com/defenseunicorns/kubernetes-fluent-client once it is fully released

Reply:
Created an issue to track: https://github.com/defenseunicorns/zarf/issues/2030